src/files/XmlFile.ts

```diff
@@ -444,14 +446,45 @@ export class XmlFile {
      * Convert the brightscript/brighterscript source code into valid brightscript
      */
     public transpile(): CodeWithSourceMap {
-        const source = this.pathAbsolute;
-        const extraImports = this.getMissingImportsForTranspile();
-        if (this.needsTranspiled || extraImports.length > 0) {
-            //emit an XML document with sourcemaps from the AST
-            return this.parser.ast.transpile(source, extraImports);
+        const state = new SGTranspileState(this);
+
+        const extraImportScripts = this.getMissingImportsForTranspile().map(uri => {
+            const script = new SGScript();
+            script.uri = util.getRokuPkgPath(uri.replace(/\.bs$/, '.brs'));
+            return script;
+        });
+
+        const extraImportIdx = this.ast.component.scripts.length - 1;
+        //temporarily add the missing imports as script tags
+        this.ast.component.scripts.push(...extraImportScripts);
```

Comment:
What about adding an internal "extraImports` property to the scripts tag and letting it transpile it?

Reply:
I don't like this idea. It's easy enough to add and remove the scripts from the `component.scripts` array. We would end up with only 1 less line of code here if we used `extraImports`, and it seems like this specific logic doesn't really belong in the AST anyway. It doesn't help plugins either, because if more than one plugin wanted to add `extraImports`, then they'd be right back to adding/removing and tracking indexes, so might as well just keep that all in the `component.scripts` collection.